src/exampleTypes/Date.js

```diff
@@ -3,22 +3,201 @@ import { observer } from 'mobx-react'
+  {
+    type: 'past',
+    label: 'Last 30 Days',
+    value: { from: 'now-30d', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 90 Days',
+    value: { from: 'now-90d', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 180 Days',
+    value: { from: 'now-180d', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 12 Months',
+    value: { from: 'now/d-12M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 24 Months',
+    value: { from: 'now/d-24M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 36 Months',
+    value: { from: 'now/d-36M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 48 Months',
+    value: { from: 'now/d-48M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 60 Months',
+    value: { from: 'now/d-60M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last Calendar Month',
+    value: { from: 'now-1M/M', to: 'now/M-1d' },
+  },
+  {
+    type: 'past',
+    label: 'Last Calendar Quarter',
+    value: { from: 'lastQuarter', to: 'lastQuarter' },
+  },
+  {
+    type: 'past',
+    label: 'Last Calendar Year',
+    value: { from: 'now-1y/y', to: 'now/y-1d' },
+  },
+  {
+    type: 'present',
+    label: 'This Calendar Month',
+    value: { from: 'now/M', to: 'now' },
+  },
+  {
+    type: 'present',
+    label: 'This Calendar Quarter',
+    value: { from: 'thisQuarter', to: 'thisQuarter' },
+  },
+  {
+    type: 'present',
+    label: 'This Calendar Year',
+    value: { from: 'now/y', to: 'now' },
+  },
+  {
+    type: 'future',
+    label: 'Next Calendar Month',
+    value: { from: 'now+1M/M', to: 'now+2M/M' },
+  },
+  {
+    type: 'future',
+    label: 'Next Calendar Quarter',
+    value: { from: 'nextQuarter', to: 'nextQuarter' },
+  },
+  {
+    type: 'future',
+    label: 'Next Calendar Year',
+    value: { from: 'now+1y/y', to: 'now+2y/y' },
+  },
+  {
+    type: 'future',
+    label: 'Next 30 Days',
+    value: { from: 'now/d', to: 'now/d+30d' },
+  },
+  {
+    type: 'future',
+    label: 'Next 12 Months',
+    value: { from: 'now/d', to: 'now/d+12M' },
+  },
+  {
+    type: 'future',
+    label: 'Next 24 Months',
+    value: { from: 'now/d', to: 'now/d+24M' },
+  },
+  {
+    type: 'future',
+    label: 'Next 36 Months',
+    value: { from: 'now/d', to: 'now/d+36M' },
+  },
+]
 
-let dateInput = x => <input type="date" {...x} />
 let DateComponent = injectTreeNode(
-  observer(({ tree, node, DateInput = dateInput }) => (
-    <Flex style={{ alignItems: 'center' }}>
-      <DateInput
-        value={node.from || ''}
-        onChange={e => tree.mutate(node.path, { from: e.target.value })}
-      />
-      <div>-</div>
-      <DateInput
-        value={node.to || ''}
-        onChange={e => tree.mutate(node.path, { to: e.target.value })}
-      />
-    </Flex>
-  )),
+  observer(
+    ({
+      tree,
+      node,
+      CustomPicker,
+      useJSPicker = true,
```

Comment:
we should not have this as a prop, we should just default the passed in date picker to the native one